libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -743,6 +746,19 @@ protected virtual void OnSetScopedServices(DialogContext dialogContext)
             if (Generator != null)
             {
                 dialogContext.Services.Set(this.Generator);
+                if (this.Generator is ResourceMultiLanguageGenerator generator)
+                {
+                    // Set generator key into state.
+                    dialogContext.State.SetValue(GeneratorIdKey, generator.ResourceId);
+
```

Comment:
NO, there is no need to do this.  This data should already be accessible via existing DialogClass memory scope aka: "dialogclass.generator.resourceId"